hyperspy/component.py

```diff
@@ -470,6 +513,32 @@ def export(self, folder=None, name=None, format=None,
             self.as_signal(field='std').save(append2pathname(
                 filename, '_std'))
 
+    def as_dictionary(self, picklable=False):
+        """Returns parameter as a dictionary, saving all attributes from self._whitelist.keys()
+                For more information see :meth:`hyperspy.misc.export_dictionary.export_to_dictionary`
+
+        Parameters
+        ----------
+        picklable : Bool (optional, False)
+            If any found functions will be pickled, and signals converted to dictionaries
```

Comment:
No clear for the user what `pickable` does. Most hyperspy users won't know what pickling is and won't be aware of the consequences of their choice here.